### PR TITLE
Ability to edit transactions

### LINKED ISSUE #84: [High][Transactions][Functional] When editing a transaction, delete confirmation popup is displayed
**URL**: http://entaxy-staging.s3-website.ca-central-1.amazonaws.com/transactions
**Environment**: Staging
**Device**: MacOS 10.13.6
**Browser**: Chrome version 68.0.3440.83
(Note: reproducible on all browsers and mobile devices)

### Steps to reproduce

1. Access transactions page
2. Add transactions
3. Click on edit icon to edit transactions

### Expected behavior

Transaction popup should be displayed to edit data.

### Actual behavior

Delete confirmation popup is displayed and upon click on "OK", no action occurs

### Screenshot

transactions

### LINKED ISSUE #90: High_Portfolio_Transaction page_Edit button confirmation page produces a duplicate transaction
### Steps to reproduce
1. Go to Portfolio page
2. Click on Transactions
3. click on existing transaction and click on edit icon (If no existing transaction go to step 4)
4. Add a new transaction at the top left (+ icon) and click on edit icon to the newly added transaction.
5. A pop up window will be displayed confirming to delete. Click Ok.

### Expected behavior
1. There should not be any delete pop up (Refer to Bug# 120)
2. When Ok button is selected either it should delete or do nothing to the transaction

### Actual behavior
1. The user should be able to edit the details in the transaction.
2. When the transaction is saved no duplicate transaction should be generated.
edit button adds a new transaction

### Screenshot
Provide a screenshot, GIF, or video of the issue if possible. People like pictures.

### LINKED ISSUE #91: High_Portfolio_Transaction_Edit button behaves like a Delete button
### Steps to reproduce
1. Click on Portfolio Page
2. Click on Transactions
3. Add a new transaction
4. Add all the mandatory fields
5. Repeat step 4 and 5 to add another transaction.
6. View the portfolio page
7. Click on edit button to the 1st transaction added.
### Expected behavior
The user should be able to edit the details of the transaction and save the transaction.

### Actual behavior
The page displays a pop up to delete the transaction.
edit button works as delete button

### Screenshot
Provide a screenshot, GIF, or video of the issue if possible. People like pictures.

### PR DESCRIPTION
**Pull request by JackNeto**
### Summary

Adding ability to edit transactions

### How to test it

Go to the transactions page and add, edit and delete transactions

### Other Information

- Fixes #84
- Fixes #90 
- Fixed #91

### CODE REVIEW

File: src/store/transactions/__tests__/types.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import _ from 'lodash'
```
Comment (pouyaemami) [inline src/store/transactions/__tests__/types.test.js]: I wonder if we should be ignoring types files as they don't really have much logic ... Let me know what you think

File: src/store/transactions/actions.js [hunk 2]
```diff
@@ -4,6 +4,10 @@ import { saveState } from '../user/actions'
 import { updatePortfolioFilters } from '../settings/actions'
 import { updateMarketValues } from '../marketValues/actions'
 
+export const loadTransactions = (transactions) => {
```
Comment (pouyaemami) [inline src/store/transactions/actions.js]: You can do a short-form return:
```javascript
export const loadTransactions = (transactions) => ({
  type: types.LOAD_TRANSACTIONS,
  payload: transactions
})

File: src/core/Portfolios/TransactionDialog/index.jsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        />
+        <TextField
+          type="number"
+          label="Shares"
+          inputProps={{
+            'aria-label': 'Shares',
+            required: true,
+            maxLength: 10,
+            min: 0,
+            max: 9999999999
```
[the remark this answers is not in the document]
Comment (JackNeto) [reply src/core/Portfolios/TransactionDialog/index.jsx]: I like `Number.MAX_SAFE_INTEGER`. I think we should have some limit on the number of character so the user doesn't type forever.

File: src/core/Portfolios/Transactions/index.jsx [hunk 4]
```diff
@@ -56,85 +58,119 @@ const mapDispatchToProps = (dispatch) => {
...
+                  {_.map(transactions, (transaction) => {
+                    return (
+                      <TableRow key={transaction.id}>
+                        <TableCell>{transaction.source}</TableCell>
+                        <TableCell>{transaction.account}</TableCell>
+                        <TableCell>{transaction.type}</TableCell>
+                        <TableCell>{transaction.ticker}</TableCell>
+                        <TableCell numeric>{transaction.shares}</TableCell>
+                        <TableCell numeric>{currencyFormatter.format(transaction.bookValue)}</TableCell>
+                        <TableCell numeric>{format(transaction.createdAt, 'MMM do, YYYY @ h:mm a')}</TableCell>
```
[the remark this answers is not in the document]
Comment (JackNeto) [reply src/core/Portfolios/Transactions/index.jsx]: Yup, we'll deal with that when we allow the user to specify the settings

**Pull request merged**